x/ibc/03-connection/keeper/keeper.go

```diff
@@ -101,33 +102,44 @@ func (k Keeper) SetClientConnectionPaths(ctx sdk.Context, clientID string, paths
 	store.Set(ibctypes.KeyClientConnections(clientID), bz)
 }
 
+// GetAllClientConnectionPaths returns all stored clients connection id paths.
+func (k Keeper) GetAllClientConnectionPaths(ctx sdk.Context) []types.ConnectionPaths {
+	var allConnectionPaths []types.ConnectionPaths
+	k.clientKeeper.IterateClients(ctx, func(cs clientexported.ClientState) bool {
+		paths, found := k.GetClientConnectionPaths(ctx, cs.GetID())
```

Comment:
Why would they not be found? Do we sometimes not set them?

Reply:
eg when you create a client but the connection handshake hasn't been initialized

Reply:
ah OK